modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java

```diff
@@ -40,6 +40,7 @@
     public String dataType;
     public String datatypeWithEnum;
     public String dataFormat;
+    public String referencedEnumDataType;
```

Comment:
Let's say we have a an enum model name E1 with these possible values: "ev1","ev2","ev3". Then we have another model named M1 with this property: enumPropery1 which is a #ref to E1, then if we want to use the dataType, we can't since it would be E1, but then how should we define E1 in PHP? The workaround is to use the type of E1 which is string here instead. This will also leave the opportunity to support a true E1 enum definition in PHP 8.1 (where real enums are supported) using direct dataType (instead of overwriting it)